Source/GTMSessionFetcher.h

```diff
@@ -441,6 +441,16 @@ extern "C" {
   #endif
 #endif
 
+#if ((defined(TARGET_OS_MACCATALYST) && TARGET_OS_MACCATALYST) || \
+     (TARGET_OS_OSX && defined(__MAC_10_12) && __MAC_OS_X_VERSION_MIN_REQUIRED >= __MAC_10_12) || \
+     (TARGET_OS_IOS && defined(__IPHONE_10_0) && __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_10_0) || \
+     (TARGET_OS_WATCH && defined(__WATCHOS_3_0) && __WATCHOS_VERSION_MIN_REQUIRED >= __WATCHOS_3_0) || \
+     (TARGET_OS_TV && defined(__TVOS_10_0) && __TVOS_VERSION_MIN_REQUIRED >= __TVOS_10_0))
+#define GTM_SUPPORTS_METRICS_COLLECTION 1
```

Comment:
This is usually necessary only for new Xcode versions, but NSURLSessionTaskMetrics has been around since iOS 10, meaning Xcode 8, and we don't support building with such an old Xcode anymore. I think we can skip this, and if necessary simply annotate methods with API_AVAILABLE to catch any attempts to call on older OS versions.

I think it would really be fine to allow setting the metricsCollectionBlock property even on older versions of the OS, as long as the property/block is documented as only being called if the OS supports metrics collection.

Reply:
> This is usually necessary only for new Xcode versions, but NSURLSessionTaskMetrics has been around since iOS 10, meaning Xcode 8, and we don't support building with such an old Xcode anymore.

I like the idea of not needing the `GTM_SUPPORTS_METRICS_COLLECTION` macro, but the problem is that `NSURLSessionTaskMetrics` is only available when building with iOS 10+ SDK, and the iOS targets in GTMSessionFetcher's Xcode project is still set to iOS 8. What would be the best practice to address this?

Reply:
The thing is, we should not have any clients at this time building with an SDK < iOS 12 (based on Apple's Xcode submission requirements), and fairly soon (IIRC) everyone should be building with the iOS 13 SDK. As a result, the symbols will exist at compile time. If we mark the property with API_AVAILABLE(ios(10.0)) etc, anyone targeting less than that would need to wrap the code in @available(). Because it's a class, we can compile with it and as long as it's not referenced from < iOS 10 code, it'll be fine.

We may need to similarly annotate one or more internal methods, but that shouldn't be problematic.

Reply:
Done. I wasn't sure about the right way to add `API_AVAILABLE` to typedefs, but Xcode showed me how.